src/Framework/AbstractFacade.php

```diff
@@ -4,7 +4,54 @@
 
 namespace Gacela\Framework;
 
+use Gacela\Framework\ClassResolver\Factory\FactoryResolver;
+use RuntimeException;
+
+/**
+ * The `__callStatic` and `__call` methods allow defining `getFactory` as static and non-static methods.
+ *
+ * @psalm-suppress MethodSignatureMismatch
+ *
+ * @method static AbstractFactory getFactory()
+ * @method AbstractFactory getFactory()
+ */
 abstract class AbstractFacade
 {
-    use FactoryResolverAwareTrait;
+    /** @var array<string, AbstractFactory> */
+    private static array $factories = [];
+
+    /**
+     * @param list<mixed> $arguments
+     */
+    public static function __callStatic(string $name = '', array $arguments = [])
+    {
+        if ($name === 'getFactory') {
+            return self::doGetFactory();
```

Comment:
Should we call the parent call and staticCall in case they exists? Does it make sense?

Reply:
I don't understand your question. For some context, check the tests here: `tests/Feature/Framework/StaticFacade/FeatureTest.php`

![Screenshot 2024-08-17 at 17 10 11](https://github.com/user-attachments/assets/2ea04527-e7bf-448c-acda-5401bca94217)